File: Algorithms/Python/SelectionSort.py
def SelectionSort(arr: list)->list:
    for i in range(len(arr)):
        min_idx = i
        for j in range(i + 1, len(arr)):
            if arr[min_idx] > arr[j]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    return arr

if __name__ == '__main__':
    a = list(map(int,  input("Enter the elements of array separated by space : ").split()))
    print("The sorted array is ",SelectionSort(a))

File: Algorithms/C++/SelectionSort.cpp
#include <bits/stdc++.h> 
using namespace std; 

void SelectionSort(int arr[], int n){ 
	int i, j, min_idx; 
	for (i = 0; i < n - 1; i++) { 
		min_idx = i; 
		for (j = i + 1; j < n; j++) { 
			if (arr[j] < arr[min_idx]) 
				min_idx = j; 
		}
		if(min_idx != i)
            swap(arr[min_idx], arr[i]); 
	} 
} 

void PrintArray(int arr[], int size){ 
	int i; 
	for (i = 0; i < size; i++) { 
		cout << arr[i] << " "; 
	}
    cout << endl; 
}

int main(){
    int n;
	cin >> n;
    int arr[n];
    for(int  i=0;i<n;i++){
        cin >> arr[i];
    }
	SelectionSort(arr, n); 
	cout << "Sorted array: \n"; 
	PrintArray(arr, n); 
	return 0; 
}